openmdao/devtools/tests/test_xdsm_viewer.py

```diff
@@ -18,21 +19,28 @@
 except ImportError:
     XDSM = None
 
+# Set DEBUG to True if you want to view the generated HTML and PDF output files.
+DEBUG = False
+# Suppress pyXDSM console output
+QUIET = not DEBUG
+
 
 @unittest.skipUnless(XDSM, "The pyXDSM package is required.")
 class TestPyXDSMViewer(unittest.TestCase):
 
     def setUp(self):
-        self.startdir = os.getcwd()
-        self.tempdir = tempfile.mkdtemp(prefix='TestPyXDSMViewer-')
-        os.chdir(self.tempdir)
+        if not DEBUG:
+            self.startdir = os.getcwd()
```

Comment:
It's not clear to me why we're not using a temp directory if DEBUG is False.  Even in the DEBUG=False case, the tests are creating local files.  These files should be cleaned up in order to avoid the possibility of tests passing when they shouldn't (due to existence of a file left over from a previous test run even if the current test failed to create it), and also just to avoid cluttering up the file system with a bunch of generated files.

Reply:
I use DEBUG=True during development to visually inspect the files (since the tests can only check if the file was created, but not that the diagram is correct). But i can get rid of it, if it is confusing or could be a source of error. Actually I stole the idea [from the problem viewer tests](https://github.com/OpenMDAO/OpenMDAO/blob/master/openmdao/devtools/problem_viewer/tests/test_viewmodeldata.py).

Reply:
Oops, my bad.  It's totally fine.  Ignore my comment.